### PR TITLE
Allow consumers to pull from multiple queues, but take effort to make su...

### PR DESCRIPTION
**Pull request by cpennington**
...re a single queue can't overload all workers

### CODE REVIEW

File: queue/management/commands/run_consumer.py [hunk 1]
```diff
@@ -1,13 +1,62 @@
...
+    if len(queue_assignments) < num_queues:
+        raise NoAssignmentError()
+
+    return dict(zip(queues, queue_assignments))
+
+
+def assign_queues_to_workers(num_workers, queue_assignments):
+    """
+    Takes a dictionary mapping queues to workers for those queues, and converts
+    it to a list of queues assigned to each worker
```
Comment (shnayder) [inline queue/management/commands/run_consumer.py]: converts to set, not list

**Comment by cpennington** [comment]
Currently takes too long to start up, trying to figure out how to speed it up

File: queue/tests/test_run_consumer.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+from itertools import combinations
+
+from queue.management.commands.run_consumer import assign_workers_to_queues, assign_queues_to_workers, NoAssignmentError
+
+
+def min_disjoint_workers(queue_assignments):
+    return min(len(set(a) - set(b)) for (a, b) in combinations(queue_assignments.values(), 2))
+
+
+def test_assign_workers_to_queues():
```
Comment (shnayder) [inline queue/tests/test_run_consumer.py:11]: Is there a jenkins job that actually runs these, btw?

Comment (cpennington) [inline queue/tests/test_run_consumer.py:11]: Yes. Just cleaned it up

File: queue/management/commands/run_consumer.py (hunk 1 above)
Comment (cpennington) [inline queue/management/commands/run_consumer.py]: Fixed

**Pull request merged**